demo-extended/src/main/java/com/novoda/downloadmanager/demo/extended/MainActivity.java

```diff
@@ -33,6 +34,19 @@ protected void onCreate(Bundle savedInstanceState) {
         com.novoda.notils.logger.simple.Log.setShowLogs(true);
         setContentView(R.layout.activity_main);
         listView = (ListView) findViewById(R.id.main_downloads_list);
+        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                DownloadAdapter adapter = (DownloadAdapter) parent.getAdapter();
+                Download item = adapter.getItem(position);
+                long batchId = item.getBatchId();
+                if (item.isPaused()) {
```

Comment:
I hate to be the party pooper but you have been randomly selected now the complexity has reached level N. What about splitting this demo into multiple activities - so that each shows a simple example of a different feature?